ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -2101,3 +2108,63 @@ - (BOOL)isEqual:(id)object {
 }
 
 @end
+
+
+#pragma mark - ORKLocationAnswerFormat
+
+@implementation ORKLocationAnswerFormat
+
+- (BOOL)isAnswerValidWithString:(nullable NSString *)text {
+    BOOL isValid = [super isAnswerValidWithString:text];
+    
+    MKPlacemark *placemark = [MKPlacemark ork_placemarkWithString:text];
+    
+    if (placemark && CLLocationCoordinate2DIsValid(placemark.coordinate)) {
+        isValid = YES;
+    } else {
+        isValid = NO;
+    }
+    
+    return isValid;
+}
+
+- (NSString *)localizedInvalidValueStringWithAnswerString:(NSString *)text {
+    if (![text length]) {
+        return nil;
+    }
+    
+    MKPlacemark *placemark = [MKPlacemark ork_placemarkWithString:text];
+    CLLocationCoordinate2D coordinate = placemark.location.coordinate;
+    NSString *string = nil;
+    
+    if (coordinate.latitude < -180.0) {
+        string = [NSString stringWithFormat:ORKLocalizedString(@"LOCATION_ALERT_MESSAGE_LATITUDE_BELOW_MINIMUM", nil), coordinate.latitude];
+    } else if (coordinate.latitude > 180.0) {
+        string = [NSString stringWithFormat:ORKLocalizedString(@"LOCATION_ALERT_MESSAGE_LATITUDE_ABOVE_MAXIMUM", nil), coordinate.latitude];
+    } else if (coordinate.longitude < -180.0) {
+        string = [NSString stringWithFormat:ORKLocalizedString(@"LOCATION_ALERT_MESSAGE_LONGITUDE_BELOW_MINIMUM", nil), coordinate.latitude];
+    } else if (coordinate.longitude > 180.0) {
+        string = [NSString stringWithFormat:ORKLocalizedString(@"LOCATION_ALERT_MESSAGE_LONGITUDE_ABOVE_MAXIMUM", nil), coordinate.latitude];
+    } else if (CLLocationCoordinate2DIsValid(coordinate)) {
+        string = [NSString stringWithFormat:ORKLocalizedString(@"RANGE_ALERT_MESSAGE_OTHER", nil), text];
+    }
```

Comment:
@umerkhan-apple It shouldn't be able to. All the coordinates are provided by the CLGeocoder, and I can probably assume that it will not return a CLPlacemark with an invalid coordinate. I added it to catch an invalid coordinate in case anything went wrong. Should I remove it?